bindings/node/lib/stateMachine.js

```diff
@@ -320,6 +320,9 @@ module.exports = function(modules) {
 
           if (request.bytesNeeded <= 0) {
             socket.end(resolve);
```

Comment:
Would it be incorrect to call the `destroySockets` helper?

Reply:
> is it a node bug or intended behavior that `.end` doesn't properly remove the handle?

I guess since we don’t pass `allowHalfOpen` to `tls.connect`, the socket should be closed once data has been written to the underlying network socket, yes.

Reply:
I've updated to use `destroy()` instead of end, removing the need of the unref.